test/e2e/e2eutil.go

```diff
@@ -29,15 +29,28 @@ func RegisterTypesWithFramework(newTypes ...runtime.Object) error {
 	return nil
 }
 
-func CreateOrUpdateMongoDB(mdb *mdbv1.MongoDB, ctx *f.TestCtx) error {
-	err := f.Global.Client.Get(context.TODO(), types.NamespacedName{Name: mdb.Name, Namespace: mdb.Namespace}, &mdbv1.MongoDB{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			return f.Global.Client.Create(context.TODO(), mdb, &f.CleanupOptions{TestContext: ctx})
+// UpdateMongoDBResource applies the provided function to the most recent version of the MongoDB resource
+// and retries when there are conflicts
+func UpdateMongoDBResource(original *mdbv1.MongoDB, updateFunc func(*mdbv1.MongoDB)) error {
```

Comment:
Can you add some sort of exponential backoff here?

Reply:
I don't think we need the backoff here, this loop is solely to account for period occurrences of apiserver errors that we get when the underlying resources are being modified elsewhere.

Reply:
In a scenario where apiserver errors are occurring, the less we want is to overwhelm the apiserver with a burst of requests.

Reply:
I think the loop is not needed here. Why would the object be modified by multiple actors?

Reply:
discussed:

- understand why there are multiple concurrent modifications, and fix that. but this 0..3 tries is tech debt.

Reply:
replaced with a single attempt, will keep track of any errors from this